### ISSUE TITLE
Migration guide?

### ISSUE THREAD

**Issue by ssuess** [nextapps-de/flexsearch#276]
Is there a clear migration guide somewhere, with examples from 0.6.3 and their equivalents in 0.7.x to help in migration?  I have poured over [this document](https://github.com/nextapps-de/flexsearch/blob/master/doc/0.7.0.md)  but it does not contain much that is helpful for a migration. For example:

- is `new Flexsearch({})` completely deprecated? (because there is still one example of it under Usage in the readme
- likewise with the new lang prefs
- since all previous indexes in the code I have were made with `new Flexsearch({})`, it is not very clear (to me anyway) if these should all now use Index or Document types. Is there a fuller explanation somewhere?

In short, I think flexsearch is amazing, but trying to adapt and migrate third party code based on 0.6.3 to 0.7.x is quite frustrating without some clear side by side examples for both index and language. Thanks!

**Comment by ssuess** [comment]
Still hoping that there is a guide somewhere, or will be? Anyone?

**Comment by Tessmore** [comment]
@ssuess I used the benchmark as an example: https://github.com/nextapps-de/flexsearch/blob/master/bench/test/flexsearch-0.7.0/index.html

* `encode: " ..."` seems to be changed, now must be a function
* ... ?

**Comment by ssuess** [comment]
@Tessmore thanks for this, but I still think an actual migration guide with side by side old/new examples doing the exact same thing is essential.

**Comment by ts-thomas** [comment]
partially yes: https://github.com/nextapps-de/flexsearch/blob/master/CHANGELOG.md

**Comment by ssuess** [comment]
I'm sorry, but unless I am missing something, the changelog is not an actual migration guide, and as far as I can tell it has not changed since I first opened this issue.
